### PR TITLE
Getting Transaction and Company Account from request session

### PR DESCRIPTION
**Pull request by ch-mcabrera**
**Filing Generator** functionality. Changes: 
- Getting Transaction and Company Account from the request session.
That will be used when calling the implementation. It will be used in the next pr
- Update junit tests

Resolves: SFA-559

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java [hunk 1]
```diff
@@ -22,23 +25,46 @@
...
     private FilingService filingService;
 
     @GetMapping("/private/transactions/{transactionId}/company-accounts/{companyAccountId}/filings")
     public ResponseEntity generateFiling(@PathParam("transactionId") String transactionId,
         @PathParam("companyAccountId") String accountId, HttpServletRequest request) {
 
+        Transaction transaction =
+            (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        if (transaction == null) {
```
Comment (ChrisJBurns) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: Isn't this stuff done in the interceptor?

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java [hunk 2]
```diff
@@ -22,23 +25,46 @@
...
+            logRequestError(request, null, "no transaction in request session");
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        CompanyAccountEntity companyAccountEntity =
```
Comment (ChrisJBurns) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: Isn't this also done in the interceptor?

File: src/main/java/uk/gov/companieshouse/api/accounts/service/FilingService.java [hunk 3]
```diff
@@ -1,6 +1,9 @@
 package uk.gov.companieshouse.api.accounts.service;
 
+import java.io.IOException;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.filing.Filing;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/FilingService.java]: Can you remove the unused imports please. Has Sonar been run against this PR?

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java [hunk 4]
```diff
@@ -3,8 +3,10 @@
 
 import org.springframework.stereotype.Service;
 import uk.gov.companieshouse.api.accounts.CompanyAccountsApplication;
+import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.filing.Filing;
 import uk.gov.companieshouse.api.accounts.service.FilingService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java]: Can you remove the unused imports please.

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java [hunk 5]
```diff
@@ -2,20 +2,30 @@
...
+@ExtendWith(MockitoExtension.class)
 @TestInstance(Lifecycle.PER_CLASS)
 public class FilingServiceImplTest {
 
     private FilingService filingService = new FilingServiceImpl();
 
+    @Mock
+    private Transaction transactionMock;
+    @Mock
+    private CompanyAccountEntity companyAccountEntityMock;
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java]: These mocks are unused.

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/FilingControllerTest.java [hunk 6]
```diff
@@ -50,7 +63,7 @@ public void shouldGenerateFiling() {
 
     @Test
     @DisplayName("Tests the unsuccessful creation of the ixbrl - filing is null")
-    void shouldNotGenerateFiling() throws IOException, NoSuchAlgorithmException {
+    void shouldNotGenerateFiling() {
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/api/accounts/controller/FilingControllerTest.java]: The unit tests are missing for the scenarios where the transaction and company account are null/not found

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java [hunk 7]
```diff
@@ -22,23 +25,46 @@
...
+            (CompanyAccountEntity) request.getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+
+        if (companyAccountEntity == null) {
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: The transaction and companyAccountEntity are not used.

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java (hunk 1 above)
Comment (ch-mcabrera) [reply src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: it should be done in the controller but the changes are not there yet. For that reason, I've added this check here.

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java (hunk 2 above)
Comment (ch-mcabrera) [reply src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: same as before

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java (hunk 1 above)
Comment (ch-mcabrera) [reply src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: It is. I will remove it.

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java (hunk 7 above)
Comment (ch-mcabrera) [reply src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java]: 43107d8

File: src/test/java/uk/gov/companieshouse/api/accounts/controller/FilingControllerTest.java (hunk 6 above)
Comment (ch-mcabrera) [reply src/test/java/uk/gov/companieshouse/api/accounts/controller/FilingControllerTest.java]: 43107d8

File: src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java (hunk 5 above)
Comment (ch-mcabrera) [reply src/test/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImplTest.java]: yes, they are used now.

File: src/main/java/uk/gov/companieshouse/api/accounts/service/FilingService.java (hunk 3 above)
Comment (ch-mcabrera) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/FilingService.java]: 43107d8

File: src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java (hunk 4 above)
Comment (ch-mcabrera) [reply src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java]: 43107d8

**Pull request merged**